native-sql-engine/cpp/src/CMakeLists.txt

```diff
@@ -156,7 +156,9 @@ macro(build_arrow STATIC_ARROW)
                       -DARROW_FILESYSTEM=ON
                       -DARROW_JSON=ON
                       -DARROW_FLIGHT=OFF
-                      -DARROW_JEMALLOC=OFF
+                      -DARROW_JEMALLOC=ON
+                      -DARROW_SIMD_LEVEL=AVX512
+                      -DARROW_RUNTIME_SIMD_LEVEL=AVX512
```

Comment:
looks like github action is running on some old servers, we may need to lower this to avx2